### ISSUE TITLE
Drawing works in simulator but not on physical iOS devices[BUG] 

### ISSUE THREAD

**Issue by twinharbor** [mono/SkiaSharp#1026]
I've run into a strange issue where I'm using an SKCanvasView to collect some simple drawing. This works fine in the simulator, but when launching on physical iOS devices the TouchAction event handler is never hit. It works fine on Android. 

This appears to be the same as issue [341](https://github.com/mono/SkiaSharp/issues/341) but that one the person who ran into it reformatted their mac and setup a whole new build environment and then it was fine. To try to replicate that, I setup everything on a separate machine but it must be tied to these specific versions or my project or something because the same result happened on a separate dev machine. 

Thanks in advance for any assistance! 

Here's the relevant code: 

```

<ContentPage.Content> 
        <RelativeLayout VerticalOptions="FillAndExpand" HorizontalOptions="FillAndExpand">
              <Image x:Name="imgBG"
                      Aspect="AspectFit"
            RelativeLayout.WidthConstraint= "{ConstraintExpression Type=RelativeToParent, Property=Width}"
            RelativeLayout.HeightConstraint= "{ConstraintExpression Type=RelativeToParent, Property=Height}"></Image>
             <StackLayout VerticalOptions="FillAndExpand" HorizontalOptions="FillAndExpand"  RelativeLayout.WidthConstraint=
         "{ConstraintExpression Type=RelativeToParent,
                                Property=Width,
                                Factor=1}"
     RelativeLayout.HeightConstraint=
         "{ConstraintExpression Type=RelativeToParent,
                                Property=Height,
                                Factor=1}" BackgroundColor="Transparent">
            <Grid VerticalOptions="FillAndExpand" HorizontalOptions="FillAndExpand" BackgroundColor="Transparent">
               <skia:SKCanvasView x:Name="canvasView" BackgroundColor="Transparent" PaintSurface="OnCanvasViewPaintSurface" VerticalOptions="FillAndExpand" HorizontalOptions="FillAndExpand" />
                <Grid.Effects>
                 <tt1:TouchEffect Capture="True"
                                        TouchAction="OnTouchEffectAction" />
            </Grid.Effects>
                </Grid>

            <Grid HorizontalOptions="CenterAndExpand" Padding="20">
             <Button x:Name="btnClear" Text="Clear" Clicked="btnClear_ClickedAsync" Grid.Column="0" BackgroundColor="{DynamicResource btnMutedBG}" TextColor="{DynamicResource btnMutedText}"></Button>   
             <Button x:Name="btnCancel" Text="Cancel" Clicked="btnCancel_ClickedAsync" Grid.Column="1" BackgroundColor="{DynamicResource btnMutedBG}" TextColor="{DynamicResource btnMutedText}"></Button>
             <Button x:Name="btnDone" Text="Done" Clicked="btnDone_ClickedAsync" Grid.Column="2" BackgroundColor="{DynamicResource btnPrimaryBG}" TextColor="{DynamicResource btnPrimaryText}"></Button>   
           </Grid>

            
        </StackLayout>
        </RelativeLayout>
       

    </ContentPage.Content>
```

Issue: OnTouchEffectAction handler is never hit when on physical iPad. 

- Version with issue: SkiaSharp.Views.Forms 1.68.0
- Last known good version: unknown
- IDE:  Visual Studio for Mac 8.3.9 (build 2)
- Platform Target Frameworks: iOS on iPad, tested on 12.4
- Works perfectly in simulator
  - Android: NA
  - iOS:  12.4
- Target Devices:  iPad air 2

**Comment by mattleibow** [comment]
It might be that the `TouchEffect` is having issues. Is this a custom effect that you created? If possible, try using the native `Touch` event on the `SKCanvasView`. Also set the `EnableTouchEvents` to true.

Docs:
https://docs.microsoft.com/dotnet/api/skiasharp.views.forms.skcanvasview.touch
https://docs.microsoft.com/dotnet/api/skiasharp.views.forms.skcanvasview.enabletouchevents

**Comment by ilsoftltd** [comment]
I had the same problem. App which used to work and then stopped working on the physical device but was on on the emulator and Android devices. Pulled my hair out trying to find out what it was. 
In the end, in Visual Studio, I used  the dependencies/references analyser. Quite a few were shown as "Unused". I deleted all these. The iOS project then didn't show as having a reference to SkiaSharp.Views.Forms so I added it....and it started working again! I think is was some sort of mis-match of DLLs, also indicated by the fact that if, when it's not auto calling the ondraw, you try and create a Paint structure for the canvas using a button handler, you receive a "DLLNotFound" exception. So I guess it's update every NuGet to the latest version, delete unused references/dependencies and double check required references are included. Hope this works for you.

**Comment by twinharbor** [comment]
Thanks for the replies! 
@ilsoftltd - thanks for this, sounds promising! Where is the dependency/reference analyzer? I'm using Visual Studio for Mac (latest version). 
@mattleibow - thanks, I do believe I have a custom effect but I think it's straight out of the example from MS (the finger painting one here: [https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/graphics/skiasharp/paths/finger-paint](url)). I'll look into your recommendation as well and see.

**Comment by ilsoftltd** [comment]
I right clicked the "References" folder under each project in turn and chose "Optimize References..," from the popup menu.
This brought up a window listing Unused and Used references and allowed me to delete the unused ones. I see there is a "Delete Unused References" menu item too which I'm guessing shortcuts this.
This was on VS2019 on Windows mind you!

**Comment by twinharbor** [comment]
Ah ha- seems some things are in VS for windows and not mac. Fortunately I also have windows running in parallels, but I've been on vs 2017. I'm going to update to 2019 and then see if I can open the project on that side. Thanks!

**Comment by twinharbor** [comment]
@ilsoftltd - do you have an addin or something? I don't have an optimize references option in vs.net 2019 for windows. Googling shows something called resharper that has a function with that name, maybe that's what you have?

**Comment by ilsoftltd** [comment]
Ah yes I do use Resharper :)

**Comment by twinharbor** [comment]
Thanks to all who posted ideas. I tried several things, and nothing worked but as I looked closer at your mention of EnableTouchEvents (@mattleibow ) I saw that in the sample code I was working from, it adds a toucheffect to Grid.Effects, whereas it seems we should be adding a touch handler directly to the SKCanvasView: EnableTouchEvents="True" Touch="CanvasView_Touch". I had to rework the handler a little bit, but overall the functionality is the same, and now it's all working! Thanks all!
